File: IA/reconhecimento_img.py
import google.generativeai as genai
import PIL.Image
import dotenv
import os

dotenv.load_dotenv(dotenv.find_dotenv())

genai.configure(api_key=os.getenv("chave_api_gemini"))

def redimensiona_img(img):
    larg, alt = img.size
    nova_larg = larg // 2
    nova_alt = alt // 2
    imagem_redimensionada = img.resize((nova_larg, nova_alt))
    return imagem_redimensionada

def verifica_e_redimensiona(img):
    larg, alt = img.size
    if larg >= 3600 or alt >= 2000:
        return redimensiona_img(img)
    else:
        return img

#inserção de imagens de teste
source = 'Captura de tela 2024-04-29 195956.png'
img_original = PIL.Image.open(source)
img_processada = verifica_e_redimensiona(img_original)

model = genai.GenerativeModel('gemini-pro-vision')
resposta = model.generate_content(["Write the classification of the food in the image(carbo,protein,candy, fruit or vegetable), just 1 word, if there ar no food in the image just write nfound",img_processada])

print(resposta.text)

File: backend/reconhecimento_img.js
import { ImageResizer } from 'react-native-image-resizer';
import { GoogleGenerativeAI } from  '@google/generative-ai';

require("dotenv").config();

const API_KEY = process.env.API_KEY;

const genAI = new GoogleGenerativeAI(API_KEY);

function redimensionaImg(img) {
  const { width: larg, height: alt } = img;

  const novaLarg = larg / 2;
  const novaAlt = alt / 2;

  return ImageResizer.createResizedImage(img.uri, novaLarg, novaAlt, 'PNG', 80);
}

async function verificaERedimensiona(img) {
  const { width: larg, height: alt } = img;

  if (larg >= 3600 || alt >= 2000) {
    return await redimensionaImg(img);
  } else {
    return img;
  }
}

const source = 'baixados.png';
const imgOriginal = { uri: source };
const imgRedimensionada = verificaERedimensiona(imgOriginal);


async function run() {
  const model = genAI.getGenerativeModel({ model: "gemini-pro-vision"});
  
  const prompt = "Write a story about a magic backpack."
  
  const result = await model.generateContent(prompt);
  const response = await result.response;
  const text = response.text();
  console.log(text);
}